tests/phpunit/tests/option/option.php

```diff
@@ -370,193 +370,329 @@ public function data_option_autoloading() {
+			'(float) 0.0'        => array( 0.0, 0.0 ),
+			'empty array'        => array( array(), array() ),
+			'false'              => array( false, false ),
 
 			/*
-			 * Non scalar values.
-			 * Loosely equal non-scalar values should almost always result in an update.
+			 * false and null are not included in these datasets
+			 * because false is the default value, which triggers
+			 * a call to add_option().
+			 *
+			 * See data_stored_as_empty_string() and its related test.
 			 */
-			array( false, array(), true ),
-			array( 'false', array(), true ),
-			array( '', array(), true ),
-			array( 0, array(), true ),
-			array( '0', array(), true ),
-			array( false, null ), // Does not update.
-			array( 'false', null, true ),
-			array( '', null ), // Does not update.
-			array( 0, null, true ),
-			array( '0', null, true ),
-			array( array(), false, true ),
-			array( array(), 'false', true ),
-			array( array(), '', true ),
-			array( array(), 0, true ),
-			array( array(), '0', true ),
-			array( array(), null, true ),
-			array( null, false ), // Does not update.
-			array( null, 'false', true ),
-			array( null, '' ), // Does not update.
-			array( null, 0, true ),
-			array( null, '0', true ),
-			array( null, array(), true ),
 		);
 	}
 
 	/**
-	 * Tests that update_option() stores an option that uses
-	 * an unfiltered default value of (bool) false.
+	 * Tests that update_option() adds a non-existent option when the new value
+	 * is stored as an empty string and false is the default value for the option.
 	 *
 	 * @ticket 22192
 	 *
-	 * @covers ::update_option
+	 * @dataProvider data_stored_as_empty_string
+	 *
+	 * @param mixed $new_value A value that casts to an empty string.
 	 */
-	public function test_update_option_should_store_option_with_default_value_false() {
+	public function test_update_option_should_add_option_when_the_new_value_is_stored_as_an_empty_string_and_matches_default_value_false( $new_value ) {
 		global $wpdb;
 
-		$option = 'update_option_default_false';
-		update_option( $option, false );
+		$this->assertTrue( update_option( 'foo', $new_value ), 'update_option() should have returned true.' );
 
-		$actual = $wpdb->query(
-			$wpdb->prepare(
-				"SELECT option_name FROM $wpdb->options WHERE option_name = %s LIMIT 1",
-				$option
-			)
-		);
+		$actual = $wpdb->get_row( "SELECT option_value FROM $wpdb->options WHERE option_name = 'foo' LIMIT 1" );
 
-		$this->assertSame( 1, $actual );
+		$this->assertIsObject( $actual, 'The option was not added to the database.' );
+		$this->assertObjectHasProperty( 'option_value', $actual, 'The "option_value" property was not included.' );
+		$this->assertSame( '', $actual->option_value, 'The value was not stored as an empty string.' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_stored_as_empty_string() {
+		return array(
+			'empty string' => array( '' ),
+			'null'         => array( null ),
```

Comment:
Per the above, I think we should also include `false` here. In the database, `false` becomes `''`, so it is not the same as the default value `false`.